### PR TITLE
[CI] Parity between requirements.txt and setup.py

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements/tune/requirements_tune.txt [hunk 1]
```diff
@@ -29,6 +29,7 @@ scikit-learn==0.24.2
 scikit-optimize==0.8.1
 sigopt==5.7.0
 smart_open==4.2.0
+tensorflow==2.4.2
```
Comment (amogkam) [inline python/requirements/tune/requirements_tune.txt:32]:
```suggestion
tensorflow==2.4.1
```

**Comment by richardliaw** [comment]
Can we close?

**Comment by amogkam** [comment]
Converted it to a draft. We should still eventually merge this though- I will fix the failing tests and resolve the merge conflicts later.

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Pull request closed without merging**